src/thing_traps.c

```diff
@@ -231,8 +231,18 @@ TbBool update_trap_trigger_line_of_sight_90_on_subtile(struct Thing *traptng, Ma
             {
                 if (creature_available_for_trap_trigger(thing))
                 {
-                    activate_trap(traptng, thing);
-                    return true;
+                    if (creature_affected_by_spell(thing, SplK_Invisibility))
```

Comment:
Use the creature_is_invisible function.

Your current code will not see invisible creatures that are temporary revealed through force visible.